RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc

```diff
@@ -90,44 +117,57 @@ GlobalPoint RecHitTools::getPosition(const DetId& id) const {
 
 int RecHitTools::zside(const DetId& id) const {
   int zside = 0;
-  if( id.det() == DetId::Forward) {
-    const HGCalDetId hid(id);
-    zside = hid.zside();
+  if (id.det() == DetId::Forward) {
+    zside = HGCalDetId(id).zside();
   } else if( id.det() == DetId::Hcal && id.subdetId() == HcalEndcap) {
-    const HcalDetId hcid(id);
-    zside = hcid.zside();
+    zside = HcalDetId(id).zside();
+  } else if (id.det() == DetId::HGCalEE || id.det() == DetId::HGCalHSi) {
+    zside = HGCSiliconDetId(id).zside();
+  } else if (id.det() == DetId::HGCalHSc) {
+    zside = HGCScintillatorDetId(id).zside();
   }
   return zside;
 }
 
 std::float_t RecHitTools::getSiThickness(const DetId& id) const {
   auto geom = geom_->getSubdetectorGeometry(id);
   check_geom(geom);
-  if( id.det() != DetId::Forward ) {
+  std::float_t thick(0.37);
+  if (id.det() == DetId::Forward) {
+    const HGCalDetId hid(id);
+    auto ddd = get_ddd(geom,hid);
+    unsigned int wafer = hid.wafer();
+    int tidx = ddd->waferTypeL(wafer);
+    thick    = idx_to_thickness*tidx;
+  } else if (id.det() == DetId::HGCalEE || id.det() == DetId::HGCalHSi) {
+    const HGCSiliconDetId hid(id);
+    auto ddd = get_ddd(geom,hid);
+    thick    = ddd->cellThick(hid.layer(),hid.waferU(),hid.waferV());
```

Comment:
if getSiThickness is used very often, it's better to store its record in an event setup

Reply:
Thickness depends on the wafer position. Again I am not changing the coding style there.